packages/mobile_app/lib/screens/registration/screen.dart

```diff
@@ -0,0 +1,226 @@
+                    fontSize: 16,
+                  ),
+                ),
+                Text(
+                  "Horoscope",
+                  style: TextStyle(
+                    color: AppColors.accent,
+                    fontSize: 16,
+                  ),
+                ),
+              ],
+            ),
+            Align(
+              alignment: Alignment.center,
+              child: Text(
+                "offline",
+                style: TextStyle(color: AppColors.textDisabled),
+              ),
+            ),
+            UserInfo(
+              name: UserInfoField(
+                  textObject: name,
+                  maxLength: 100,
+                  hint: lang.name,
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${lang.atLeastXsymbolsNeeded} $min";
+                    return null;
+                  }),
+              month: UserInfoField(
+                  textObject: month,
+                  maxLength: 2,
+                  keyboardType: TextInputType.datetime,
+                  hint: "mm",
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    if (int.parse(text) > 12 || int.parse(text) < 1) return "x";
+                    return null;
+                  }),
+              day: UserInfoField(
+                  textObject: day,
+                  maxLength: 2,
+                  keyboardType: TextInputType.datetime,
+                  hint: "dd",
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    if (int.parse(text) > 31 || int.parse(text) < 1) return "x";
+                    return null;
+                  }),
+              year: UserInfoField(
+                  textObject: year,
+                  maxLength: 4,
+                  keyboardType: TextInputType.datetime,
+                  hint: "yyyy",
+                  validator: (String text) {
+                    int min = 4;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    return null;
+                  }),
+              sex: UserInfoPicker(
+                items: indexToSex,
+                indexObject: sex,
+              ),
+              country: UserInfoField(
+                  textObject: country,
+                  maxLength: 50,
+                  hint: lang.birthcountry,
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${lang.atLeastXsymbolsNeeded} $min";
```

Comment:
I guess this validator should be changed (as well as for all text fields here). 
1. This field is not mandatory
2. I dont think we must indicate now many characters left

Reply:
But we don't want to break DateTime, do we?
It is better to validate only one form of writting date. The user can write 1 or 01, also user can write 600 year of birth.
And if the first example is OK, then second one can give the UTC clock a heartattack.